src/worker/plugins/run.ts

```diff
@@ -66,6 +81,16 @@ export async function runPluginsOnBatch(server: PluginsServer, batch: PluginEven
             }
         }
 
+        for (const event of returnedEvents) {
+            if ((event && pluginsRan.length > 0) || pluginsFailed.length > 0) {
```

Comment:
this `if` is looking funny... :)

Reply:
huh formatter messed up there

Reply:
Did it now? :P 
Or did you just forget the `()` and prettier just made the default operator precedence apparent? :)
Always when mixing `&&` and `||` put as many brackets around the different parts of the query as possible to avoid any errors...